src/datasources/errors/http-error-factory.ts

```diff
@@ -32,6 +32,8 @@ export class HttpErrorFactory {
 function isNetworkResponseError(
   error: NetworkError | Error,
 ): error is NetworkResponseError {
-  const responseError = error as NetworkResponseError;
-  return responseError.status >= 400 && responseError.status < 600;
+  if ('status' in error) {
```

Comment:
I think we could be simplyfied even further:
```
return 'status' in error && error.status >= 400 && error.status < 600;
```

Reply:
Great suggestion. I've change it in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.